pkg/gateway/db/pq/pq.go

```diff
@@ -0,0 +1,69 @@
+package pq
+
+import (
+	"fmt"
+	"context"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+
+	"github.com/skygeario/skygear-server/pkg/gateway/db"
+)
+
+// Open returns a new connection to postgresql implementation
+func Open(ctx context.Context, connString string) (db.Conn, error) {
+	db, err := getDB(connString)
```

Comment:
In the case of gateway, the app is only connecting one single database, so I think the `Open` function can be as simple as directly calling `sqlx.Open("postgres", req.connString)`.

The `dbInitializer` function imo is meant to handle multiple connection across multiple databases, which is overkill in the gateway.